scripts/restart-nix-daemon.sh

```diff
@@ -8,10 +8,8 @@ if [[ "$RUNNER_OS" == "Linux" ]]; then
   sudo systemctl daemon-reload
   sudo systemctl restart nix-daemon.service
 elif [[ "$RUNNER_OS" == "macOS" ]]; then
-  sudo launchctl unload /Library/LaunchDaemons/org.nixos.nix-daemon.plist
-  sudo launchctl load /Library/LaunchDaemons/org.nixos.nix-daemon.plist
-  # for some reason running this twice restarts the service
-  sudo launchctl kickstart -k -p system/org.nixos.nix-daemon
+  sudo launchctl bootout system /Library/LaunchDaemons/org.nixos.nix-daemon.plist
+  sudo launchctl bootstrap system /Library/LaunchDaemons/org.nixos.nix-daemon.plist
   sudo launchctl kickstart -k -p system/org.nixos.nix-daemon
```

Comment:
```suggestion
  # running kickstart twice ensures that we wait for service to be restarted
  sudo launchctl kickstart -k -p system/org.nixos.nix-daemon
  sudo launchctl kickstart -k -p system/org.nixos.nix-daemon
```

Reply:
I've "improved" the comment from before to make it clear why we need to run kickstart two times.

Reply:
But, that's a bit counterintuitive. launchctl has to wait for the process to spawn, since we're asking launchctl to also print the process' PID with `-p`. Just out of pure curiosity, since I wasn't able to reproduce the bug that the comment suggests, where can I find more info about the double kickstart?

Reply:
This was from my testing when I was working on `install-flox-action`. I don't remember where I got the idea to just run the kickstart two times, but I got it from somewhere. Also this was not something you are able to reproduce every time, but only happened in 10-20% of the cases.

When PID is returned the service is not *yet* up. If you test it yourself you will notice that the first return of the command is almost immediate. But when running it for the second time you will see a normal delay (for a second or two) when you restart nix-daemon service.